pandas/tests/resample/test_datetime_index.py

```diff
@@ -1816,3 +1816,14 @@ def test_resample_aggregate_functions_min_count(func):
         index=DatetimeIndex(["2020-03-31"], dtype="datetime64[ns]", freq="Q-DEC"),
     )
     tm.assert_series_equal(result, expected)
+
+
+def timeseries_interpolation():
```

Comment:
this will not run as a test. pytest will run all files of the form test_*.py or *_test.py

rename this `test_timeseries_interpolation`

Reply:
I think I have done 90 % of the job, now I let maintainers finishing it. It should take them minutes while it would take my hours to polish the PR.

This bug has been reported 3 years ago and nobody from Pandas tried anything while a solution was proposed in the bug report. Now, 3 years after, I have done a PR. I know it is not perfect but it must be very close to what it should be. I hope it will not take 3 more years.

@attack68 It is not against you. I don't want to become a maintainers but just to help at my level.

Reply:
@attack68 or any one else: feel free to for from my work and to finish the PR.

Reply:
@oricou the main thing you need to do at this point is rename this function as @attack68 suggested.

> I think I have done 90 % of the job

there are 58k lines of code in pandas/core/ and 220k in pandas/tests/.  proper testing is important.